contracts/upgrader/tests/atomic_upgrades.rs

```diff
@@ -47,6 +48,38 @@ fn upgrade_and_migrate_are_atomic() {
     });
 }
 
+#[test]
+fn upgrade_fails_if_upgrading_to_the_same_version() {
+    let env = Env::default();
+
+    let upgrader_address = env.register(Upgrader, ());
+
+    let contract_owner = Address::generate(&env);
+    let contract_address = env.register(DummyContract, (&contract_owner,));
+    let dummy_hash = BytesN::from_array(&env, &[1u8; 32]);
+    let dummy_data = String::from_str(&env, "");
+    let original_version = String::from_str(&env, "0.1.0");
+
+    let dummy_client = DummyContractClient::new(&env, &contract_address);
+
+    let upgrader = UpgraderClient::new(&env, &upgrader_address);
+
+    let upgrade_auth = mock_auth!(env, contract_owner, dummy_client.upgrade(dummy_hash));
+    let migrate_auth = mock_auth!(env, contract_owner, dummy_client.migrate(dummy_data));
+
+    assert_contract_err!(
+        upgrader
+            .mock_auths(&[upgrade_auth, migrate_auth])
```

Comment:
```suggestion
            .mock_all_auths_allowing_non_root_auth()
```
instead of mock_all_auths